### ISSUE TITLE
GBDT model trained on data already exposed to LSTM

### ISSUE THREAD

**Issue by EllieRoseS** [pinkeshbadjatiya/twitter-hatespeech#8]
Hi, it appears that the huge boost in performance that is achieved by running the GBDT over the weights from the LSTM model is due to the fact that the train/test splits are not the same in the two cases. Hence when evaluating the outcome of the GBDT, comments from the test set have already been exposed to the LSTM during training. If I keep the same train/test split when training both models I am unable to replicate your results.

**Comment by punyajoy** [comment]
Hi, I have also identified this issue with the LSTM+GBDT model. Your work is very interesting and is already a important part of hate speech research.  But the exposure to test set at the time of training will not be helpful in analyzing the real domain of hate speech.

**Comment by viniciusmss** [comment]
I agree with the above. For the curious trying to attempt to replicate this paper, I have an update code base with many bugs fixed and provide a replication summary [here](https://github.com/viniciusmss/Hate-Compare/tree/master/Replications). Feel free to shoot me a message if you're trying to replicate their results and need help.

**Comment by amittos** [comment]
@viniciusmss That's awesome, thank you very much for doing this and sharing the updated versions with us.
